plugins/copr.py

```diff
@@ -507,6 +509,38 @@ def _download_repo(self, project_name, repo_filename):
         shutil.copy2(f.name, repo_filename)
         os.chmod(repo_filename, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH)
 
+    def _resolve_error_message(self, project_name):
+        list_chroots = requests.get(self.copr_url + "/api_3/mock-chroots/list")
+        project_name_items = project_name.split("/")
+        user = project_name_items[0]
+        project = project_name_items[1]
+        params = {"ownername": user, "projectname": project}
+        project = requests.get(self.copr_url + "/api_3/project", params=params)
+        list_builds = requests.get(self.copr_url + "/api_3/build/list", params=params)
```

Comment:
This means several API queries for each repo file miss ... could the original error 404 page be modified to provide all the necessary info, and we could just present that here?

I know that the repo file is not a part of our API, though in future it could be?

Note that it would be nice to see the info on the page itself, e.g.:
https://copr.fedorainfracloud.org/coprs/praiskup/ping/repo/epel-6/praiskup-ping-epel-6.repo?arch=ppc64le

Reply:
It is only during a failure. So very rarely. I do not think we need to optimize here. I would rather prefer clean code over speed.

Reply:
I agree with @praiskup, those requests take time and the information could be displayed directly on the 404 page. I could add a repo file to the API.

Reply:
If you found an elegant way to read the "error message" from the current web page, I wouldn't be against keeping it as is (without an additional API route).  Perhaps we could even provide some additional HTTP header to avoid the HTML parsing?

Reply:
Do you mean that a header would contain the error message or at least part of the error message?

Reply:
Yes, e.g. part of the message.  Or even e.g. some json data, like `{'available_chroots': ['a', 'b', 'c']}` (we could easily construct useful message from such data here).

Reply:
See lines
```
msg = base64.b64decode(response.headers["Error-Message"]).decode('utf-8')
error_msg = ''.join(["It wasn't possible to enable this project.\n", msg])
```
The `Error-Message` header contains the whole error message which is intended just for cli, for HTML another message is displayed to users.